### PR TITLE
[BC-breaking] Unified input for F.perspective

### PR DESCRIPTION
**Pull request by vfdev-5**
Related to #2292

Description:
-  Unified input for F.perspective
- [x] added tests
- [x] updated docs

BC-breaking change because we changed the default value of `interpolate` to be `BILINEAR` instead of `BICUBIC` for `perspective`

### CODE REVIEW

File: torchvision/transforms/functional.py [hunk 1]
```diff
@@ -491,53 +491,76 @@ def hflip(img: Tensor) -> Tensor:
...
 
-    A = torch.tensor(matrix, dtype=torch.float)
-    B = torch.tensor(startpoints, dtype=torch.float).view(8)
-    res = torch.lstsq(B, A)[0]
-    return res.squeeze_(1).tolist()
+    b_matrix = torch.tensor(startpoints, dtype=torch.float).view(8)
+    res = torch.lstsq(b_matrix, a_matrix)[0]
+    # We have to explicitly produce the list of floats, otherwise torch.jit.script does recognize output type
+    # RuntimeError: Expected type hint for result of tolist()
+    return [float(i.item()) for i in res[:, 0]]
```
Comment (fmassa) [inline torchvision/transforms/functional.py:521]: I believe you can do something like
```python
output : List[float] = res.squeeze(1).tolist()
return output
```

File: torchvision/transforms/functional.py [hunk 2]
```diff
@@ -491,53 +491,76 @@ def hflip(img: Tensor) -> Tensor:
...
-    opts = _parse_fill(fill, img, '5.0.0')
+    if not isinstance(img, torch.Tensor):
+        return F_pil.perspective(img, coeffs, interpolation=interpolation, fill=fill)
 
-    coeffs = _get_perspective_coeffs(startpoints, endpoints)
-    return img.transform(img.size, Image.PERSPECTIVE, coeffs, interpolation, **opts)
+    if interpolation == Image.BICUBIC:
+        # bicubic is not supported by pytorch
+        # set to bilinear interpolation
+        interpolation = 2
```
Comment (fmassa) [inline torchvision/transforms/functional.py:561]: I'm not sure that silently changing the value requested by the user is a good thing to do here.

It is unfortunate that bicubic is not supported in PyTorch yet, but I think we should try to be consistent here (the user requested bicubic and we silently compute bilinear).

What about one of the following:
1. change the default value for the whole function to be `BILINEAR`, and raise an error if BICUBIC is passed by the user, and add a `[BC-breaking]` in the tile of the PR
2. raise a warning before switching the default value here to `bilinear`

I think I would prefer option 1 here, as option 2 would by default raise warnings to the user.

File: torchvision/transforms/functional.py (hunk 1 above)
Comment (vfdev-5) [reply torchvision/transforms/functional.py:521]: Sounds good, especially that we have min pyhon 3.6

File: torchvision/transforms/functional.py (hunk 2 above)
Comment (vfdev-5) [reply torchvision/transforms/functional.py:561]: I agree, that silent changing the value is not good at all ! It was a full-moon or something :)
I'll remove this part of code. Error is raised inside `F_t.affine` if resample is not support: 
https://github.com/pytorch/vision/pull/2558/files#diff-77c635250f0215f979b2650c383ecbecL637

**Pull request merged**